pkg/controller/registryinfra/registryinfra_controller.go

```diff
@@ -0,0 +1,840 @@
+		return nil
+	}
+
+	//else deprovision
+	err = c.deprovisionS3Bucket(bucketName, awsClient)
+	if err != nil {
+		return fmt.Errorf("error deleting bucket: %v", err)
+	}
+
+	err = deprovisionS3IamUser(clusterID, bucketName, awsClient)
+	if err != nil {
+		return fmt.Errorf("error deleting registry IAM user: %v", err)
+	}
+
+	err = c.deleteRegistyUserSecret(objectStoreSecretName, cluster.Namespace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteRegistryUserSecret will delete secret holding IAM registry user's cloud creds
+func (c *Controller) deleteRegistyUserSecret(secretName string, secretNamespace string) error {
+	_, err := c.kubeClient.CoreV1().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error querying for existing registry secret: %v", err)
+	} else if err == nil {
+		// found existing secret
+		err = c.kubeClient.CoreV1().Secrets(secretNamespace).Delete(secretName, &metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("error deleting registry secret: %v", err)
+		}
+	}
+	return nil
+}
+
+// setupRegistrySecret will save IAM registry user's cloud creds into secret for later
+// use when setting up the cluster registry
+func (c *Controller) setupRegistrySecret(secretName string, secretNamespace string, accessKey *iam.AccessKey) error {
+
+	if accessKey == nil {
+		return fmt.Errorf("accesskey data is nil. cannot save registry user's secret")
+	}
+
+	_, err := c.kubeClient.CoreV1().Secrets(secretNamespace).Create(&kapi.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+		},
+		StringData: map[string]string{
+			"awsAccessKeyId":     *accessKey.AccessKeyId,
+			"awsSecretAccessKey": *accessKey.SecretAccessKey,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error saving aws creds into registry secret: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Controller) updateClusterStatus(cluster *clusterapi.Cluster, success bool) error {
+	clusterProviderStatus, err := controller.ClusterProviderStatusFromCluster(cluster)
+	if err != nil {
+		return fmt.Errorf("error retrieving cluster status: %v", err)
+	}
+
+	newConditions := c.updateClusterCondition(clusterProviderStatus, success)
+	clusterProviderStatus.Conditions = newConditions
+
+	clusterProviderStatus.RegistryInfraInstalled = success
+
+	raw, err := controller.EncodeClusterProviderStatus(clusterProviderStatus)
+	if err != nil {
+		return fmt.Errorf("error converting clusterproviderstatus: %v", err)
+	}
+
+	clusterCopy := cluster.DeepCopy()
+	clusterCopy.Status.ProviderStatus = raw
+
+	err = controller.UpdateClusterStatus(c.capiClient, clusterCopy)
+	if err != nil {
+		return fmt.Errorf("error updating cluster status: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Controller) updateClusterCondition(clusterProviderStatus *clusteroperator.ClusterProviderStatus, suceeded bool) []clusteroperator.ClusterCondition {
```

Comment:
As mentioned above, add a condition for deprovisioning and whether it succeeded or failed, and if it failed, why it failed.